required/latex-lab/latex-lab-amsmath.dtx

```diff
@@ -75,87 +76,697 @@
+\def\maketag@@@#1
+  {%
+    \ifmeasuring@
+      \hbox{\m@th\normalfont#1}%
+    \else
+      \UseTaggingSocket{math/display/tag/begin}
+      \hbox{\m@th\normalfont#1
+      \UseTaggingSocket{math/luamml/mtable/tag/save}
+      }%
+      \UseTaggingSocket{math/display/tag/end}
+    \fi
+   }
+%    \end{macrocode}
+% \end{macro}
+% \cs{eqref} uses \cs{tagform@} and so \cs{maketag@@@} but we do not want this tagging
+% there. 
+% 
+% \begin{macro}{\eqref}
+% \begin{macro}{\maketag@@@notog}
+%    \begin{macrocode}
+\def\maketag@@@notag#1{\hbox{\m@th\normalfont#1}}
+\DeclareRobustCommand{\eqref}[1]
+  {\textup{\let\maketag@@@\maketag@@@notag\tagform@{\ref{#1}}}}
+%    \end{macrocode}
+% \end{macro}
+% \end{macro}
+%
+% \subsubsection{align \& friends}
+% 
+% Most display environment uses a common command for the end which contains
+% the luamml socket to finalize the mtable. 
+% \begin{macro}{\common@align@ending}
+%    \begin{macrocode}
+\def\common@align@ending {
+  \math@cr \black@\totwidth@
+  \UseExpandableTaggingSocket {math/luamml/mtable/finalize} {align}
+  \egroup
+  \ifingather@
+    \restorealignstate@
+    \egroup
+    \nonumber
+    \ifnum0=`{\fi\iffalse}\fi
+  \else
+    $$%
+  \fi
+  \ignorespacesafterend
+}
+%    \end{macrocode}
+% \end{macro}
+% Now we redefine the display alignments to use these ending
+% 
+%    \begin{macrocode}
+\renewenvironment{align}{%
+  \start@align\@ne\st@rredfalse\m@ne
+}{%
+  \common@align@ending
+}
+\renewenvironment{align*}{%
+  \start@align\@ne\st@rredtrue\m@ne
+}{%
+  \common@align@ending
+}
+\renewenvironment{alignat}{%
+  \start@align\z@\st@rredfalse
+}{%
+  \common@align@ending
+}
+\renewenvironment{alignat*}{%
+  \start@align\z@\st@rredtrue
+}{%
+  \common@align@ending
+}
+\renewenvironment{flalign}{%
+  \start@align\tw@\st@rredfalse\m@ne
+}{%
+  \common@align@ending
+}
+\renewenvironment{flalign*}{%
+  \start@align\tw@\st@rredtrue\m@ne
+}{%
+  \common@align@ending
+}
+\renewenvironment{xalignat}{%
+  \start@align\@ne\st@rredfalse
+}{%
+  \common@align@ending
+}
+\renewenvironment{xalignat*}{%
+  \start@align\@ne\st@rredtrue
+}{%
+  \common@align@ending
+}
+\renewenvironment{xxalignat}{%
+  \start@align\tw@\st@rredtrue
+}{%
+  \common@align@ending
+}
+%    \end{macrocode}
+% And register these environments for the math grabbing
+% 
+%    \begin{macrocode}
+  \math_register_halign_env:nn {align}{}
+  \math_register_halign_env:nn {align*}{}
+  \math_register_halign_env:nn {alignat}{}
+  \math_register_halign_env:nn {alignat*}{}
+  \math_register_halign_env:nn {flalign}{}
+  \math_register_halign_env:nn {flalign*}{}
+  \math_register_halign_env:nn {xalignat}{}
+  \math_register_halign_env:nn {xalignat*}{}
+  \math_register_halign_env:nn {xxalignat}{}
+%    \end{macrocode}
+%
+% The align preamble (used in \cs{align@}) needs code for luamml 
+% to save the cells.
+% \begin{macro}{\align@preamble}
+%    \begin{macrocode}
+\def\align@preamble{%
+   &\hfil
+    \strut@
+    \setboxz@h
+      {
```

Comment:
Missing %?

Reply:
As David this is all in ExplSyntaxOn so I will close as resolved.